frontend/readline.c

```diff
@@ -37,10 +43,39 @@ static void sigint_handler(int n) {
 }
 #endif
 
+
+static void mkdirs(const char *path) {
+	struct stat st = {0};
+	if (stat(path, &st) == 0 && S_ISDIR(st.st_mode))
+		return;
+
+	char *subpath;
+
+	subpath = strdup(path);
+	dirname(subpath);
+
+
+	if (strlen(subpath) > 1)
+		mkdirs(subpath);
+	printf("mkdir %s\n", path);
+	mkdir(path, 0700);
+	free(subpath);
+}
+
 static const char *get_history_path(void) {
 	static char history_path[PATH_MAX];
-	snprintf(history_path, sizeof(history_path),
-			"%s/.mrsh_history", getenv("HOME"));
+	if (getenv("XDG_DATA_HOME")) {
+		snprintf(history_path, sizeof(history_path),
+				"%s/mrsh", getenv("XDG_DATA_HOME"));
+	} else {
+		snprintf(history_path, sizeof(history_path),
+				"%s/.local/share/mrsh", getenv("HOME"));
+	}
+
+	mkdirs(history_path);
+
+	strncat(history_path, "/history", sizeof(history_path) - strlen(history_path) - 1);
```

Comment:
Maybe a bikeshed:

From [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html):
> Other specifications may reference this specification by specifying the location of a data file as `$XDG_DATA_DIRS/subdir/filename`. This implies that: 
> - Such file should be installed to `$datadir/subdir/filename` with `$datadir` defaulting to `/usr/share`.

From [Filesystem Hierarchy Standard, Version 3.0, 4.11. /usr/share : Architecture-independent data](https://refspecs.linuxfoundation.org/FHS_3.0/fhs/ch04s11.html):
>The `/usr/share` hierarchy is for all read-only architecture independent data files.

Obviously, the `history` file should not be read-only, so `$XDG_CACHE_DIR` would be suitable in this case.

Reply:
`XDG_DATA_DIRS` is the system-wide version of `XDG_DATA_HOME` and is only used when files which might otherwise go in `XDG_DATA_HOME` are being installed as e.g. defaults for all users.

`XDG_CACHE_HOME` is a totally unrelated concept, and refers specifically to files that may be deleted at nay time without causing loss, but might be slow to regenerate. `XDG_CACHE_DIR` does not exist, period.

Reply:
Ah, I see. Thank you for your clarification and sorry for my misunderstanding.